### PR TITLE
T9: Adds landing page

### PR DESCRIPTION
**Pull request by suzunak**
Screenshot 2024-06-01 at 7 40 07 PM

Screenshot 2024-06-01 at 7 40 20 PM

Created the landing page for Permuto. 

The following folders were added in the src folder: 
- components
- pages

The following files were edited: 
- App.js

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+      <div className="App">
+        <LandingPage />
```
Comment (justinpchen94) [inline frontend/src/App.js:6]: 🎉

File: frontend/src/App.js [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
+      </div>
   );
 }
 
 export default App;
+
+
+// import logo from './logo.svg';
```
Comment (justinpchen94) [inline frontend/src/App.js]: not sure if this is intentional but because we have source control, we don't have to worry about preserving old code "just in case". we can always look back in the source tree (git) and see what the file used to look like and undo changes if we make mistakes!

tldr - remove these comments

File: frontend/package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "@testing-library/user-event": "^13.5.0",
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
-    "react-scripts": "5.0.1",
+    "react-scripts": "^5.0.1",
```
Comment (justinpchen94) [inline frontend/package.json:11]: just curious - did you add this on purpose?

File: frontend/src/components/Body/BodyOne/BodyOne.css
Comment (justinpchen94) [inline frontend/src/components/Body/BodyOne/BodyOne.css:1]: you can refer to this page on [CSS modules](https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/) for a little more color but tldr, without changing the naming convention of this file, you will end up having all of your CSS class names shared across all components. creating a CSS module file will allow you to guarantee that the styles you use here won't collide with classes with the same name in other CSS files. it may be easy to guarantee this right now when the codebase is small, but this is really important when working in much larger codebases!

File: frontend/src/components/Body/BodyOne/BodyOne.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+// LOGO
+// IMG / BANNER
+// Description & image
```
Comment (justinpchen94) [inline frontend/src/components/Body/BodyOne/BodyOne.js]: i assume these comments were for your own notes? if so, let's remove here and in other files!

File: frontend/src/components/Body/BodyOne/BodyOne.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+import './BodyOne.css'
+
+function BodyOne() {
+    return (
+        <div className="BodyOne">
+            <header className="Logo">
```
Comment (justinpchen94) [inline frontend/src/components/Body/BodyOne/BodyOne.js]: good use of semantic html!

File: frontend/src/components/Body/BodyOne/BodyOne.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                <img src="" alt="[ LOGO ]" />
```
Comment (justinpchen94) [inline frontend/src/components/Body/BodyOne/BodyOne.js]: i love that you thought to add alt text to your images! this will go a long way with accessibility concerns in the future

however, when it comes to alt text itself, it's important to know that this is the text that will be read aloud by screen readers and also indexed by search engines. i know this is just a placeholder, but commenting this now so you know what to replace later

also in general, if you have placeholder work, it's a decently good practice to add a comment such as the following:

`// TODO (T300): replace with correct images`

where T300 represents the unique ticket number in your project tracker. this implies also that it is good practice to track these tasks down in your task tracker!

File: frontend/src/components/Body/BodyOne/BodyOne.css [hunk 7]
```diff
@@ -0,0 +1,63 @@
+.BodyOne {
```
Comment (justinpchen94) [inline frontend/src/components/Body/BodyOne/BodyOne.css]: a couple general CSS comments that i will leave here instead of in all of your CSS files:

1) good [CSS styling](https://getbem.com/) would dictate that we standardize to hyphen naming across the board. so in this case, let's do `.body-one`
2) in conjunction with 1, CSS classes are generally lower cased everywhere

File: frontend/src/components/Body/BodyThree/BodyThree.css [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+.IMG-container-B3 {
+    display: flex;
+    justify-content: space-around;
+    align-items: center;
+    padding: 50px;
+}
+
+.IMG-container-B3 img {
+    display: block;
+}
```
Comment (justinpchen94) [inline frontend/src/components/Body/BodyThree/BodyThree.css]: it feels silly but let's make sure we add a new line/empty line at the end of each file, which will get rid of these red icons

File: frontend/src/components/Body/WholeBody/WholeBody.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+import '../BodyOne/BodyOne.css';
+import '../BodyTwo/BodyTwo.css';
+import '../BodyThree/BodyThree.css';
+
+function Body() {
+    return (
+        <div>
+            <BodyOne />
+            <BodyTwo />
+            <BodyThree />
```
Comment (justinpchen94) [inline frontend/src/components/Body/WholeBody/WholeBody.js]: on naming in general, one of the important concepts in coding is the idea of [self documenting code](https://swimm.io/learn/documentation-tools/tips-for-creating-self-documenting-code#:~:text=code%20even%20clearer-,What%20is%20self%2Ddocumenting%20code%3F,the%20need%20for%20extensive%20documentation.) which is somewhat already descriptive, but in practice we want to reduce the mental overhead as much as possible for someone new (or even yourself, just further down the line) to understand your code. While it may make sense right now contextually to name these components BodyOne and such, this is actually not terribly descriptive in and of itself. It would be easier to name something like `Testimonials` or `ProductFeatures` or etc based on the content and usage of the component.

File: frontend/src/pages/landing_page/LandingPage.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import NavBar from "../../components/NavBar/NavBar";
+import WholeBody from "../../components/Body/WholeBody/WholeBody";
+import Footer from "../../components/Footer/Footer";
+
+export default function LandingPage() {
+    return (
+        <>
+            <NavBar />
+            <WholeBody />
```
Comment (justinpchen94) [inline frontend/src/pages/landing_page/LandingPage.js]: let's rename this component to something more useful but otherwise, this component is 👌

**Review by justinpchen94 — COMMENTED** [review]
looking really good for a first pr!! i did a pass on a couple things and will take a look again after you push more changes

additionally, 2 things to do at the PR level (i.e. not coding)

1) let's add the ticket number to the PR title using the `T#` syntax you see in your tracker. This makes it easy for us to tie PR's to the tasks in our task tracker
2) it is excellent practice in general whenever you make changes to any UI to include screenshots of behavior. It makes it wayyyy easier for us to review if we have an idea on how your changes look in app without having to run the code ourselves on our own computers!

File: frontend/package.json (hunk 3 above)
Comment (suzunak) [reply frontend/package.json:11]: I don't recall touching these files. Should I fix them?

Comment (justinpchen94) [reply frontend/package.json:11]: nah it's not a big deal! See this [post](https://stackoverflow.com/a/22345808) if you're curious what it means

File: frontend/src/components/Banner/banner.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import './banner.scss';
+
+function Banner() {
+    return (
+        <div className="banner">
+            {/* TODO T#: replace banner / img */}
```
Comment (justinpchen94) [inline frontend/src/components/Banner/banner.js]: almost there! Let's have you create an actual task in the tracker and then you can add the actual T number in here and other places below

**Review by justinpchen94 — APPROVED** [review]
looking really good! Feel free to merge after you update the todo comments
